src/game.cpp

```diff
@@ -11485,6 +11485,16 @@ bool game::walk_move( const tripoint &dest_loc )
     }
     u.set_underwater(false);
 
+    const tripoint furn_pos = u.pos() + u.grab_point;
+    const tripoint furn_dest = dest_loc + u.grab_point;
+    const int fire_str = m.get_field_strength( furn_pos, fd_fire );
+    const int fire_age = m.get_field_age( furn_pos, fd_fire );
+    if( m.get_field( furn_pos, fd_fire ) != nullptr ) {
+        m.remove_field( furn_pos, fd_fire );
+        m.set_field_strength( furn_dest, fd_fire, fire_str );
+        m.set_field_age( furn_dest, fd_fire, fire_age );
```

Comment:
OK, so the problem is that you replace the fire immediately. You need to wait with this until the player finishes moving, well down below, by the end of the function.

But do test it that it doesn't remove fires if player isn't dragging anything.
Another problem you need to take care of here is that it doesn't combine or delete fires too much. If you move a fire-containing stove with fire strength above 1, it should not move the fire.
So basically, you want to check if the furniture being dragged:
* Is a furniture (and not a vehicle) - check `shifting_furniture`
* Has a fire container flag - you'll need to get the furniture type object (with `m.furn`) for that
* Contains a fire of exactly 1 strength
* Is being moved into a place with at most strength 1 fire (if more than that, you can delete the original fire or leave it, just don't allow dragging oversized fires or deleting them by pushing stoves into them)